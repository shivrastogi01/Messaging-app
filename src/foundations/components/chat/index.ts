export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
