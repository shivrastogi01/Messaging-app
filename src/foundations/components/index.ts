export * from "./presentation";
export * from "./layout";
