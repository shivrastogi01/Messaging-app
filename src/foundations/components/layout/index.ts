export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
