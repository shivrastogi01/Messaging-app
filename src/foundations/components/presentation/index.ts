export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
