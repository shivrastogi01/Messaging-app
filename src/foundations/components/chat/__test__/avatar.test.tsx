import React from "react";
import { Avatar, AvatarVariants } from "foundations/components/chat";
import renderWithTheme from "main/testing/render-with-theme";

describe("Avatar", () => {
  test("default variant matches the snapshot", () => {
    const { container } = renderWithTheme(<Avatar>AV</Avatar>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("round variant matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Avatar variant={AvatarVariants.ROUND}>RA</Avatar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("with string color/bg matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Avatar color="red" bg="black">
        CA
      </Avatar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
