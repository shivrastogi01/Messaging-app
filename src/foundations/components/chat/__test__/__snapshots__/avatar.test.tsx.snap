// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar default variant matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F7861C;
  border-radius: 10px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  height: 36px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 12px;
  text-transform: uppercase;
  width: 36px;
}

<div
  class="c0"
>
  AV
</div>
`;

exports[`Avatar round variant matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F7861C;
  border-radius: 10px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  height: 36px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 12px;
  text-transform: uppercase;
  width: 36px;
  border-radius: 100vmax;
}

<div
  class="c0"
>
  RA
</div>
`;

exports[`Avatar with string color/bg matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F7861C;
  border-radius: 10px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  height: 36px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 12px;
  text-transform: uppercase;
  width: 36px;
  color: red;
  background-color: black;
}

<div
  class="c0"
  color="red"
>
  CA
</div>
`;
